File: core/src/main/scala/com/graphbrain/db/EntityNode.scala
package com.graphbrain.db

case class EntityNode(override val id: String,
                    override val degree: Int = 0,
                    override val ts: Long = -1)
  extends Vertex(id, degree, ts) {

  def this(id: String, map: Map[String, String]) =
    this(id,
      map("degree").toInt,
      map("ts").toLong)

  override def extraMap = Map()

  override def setId(newId: String): Vertex = copy(id=newId)

  override def setDegree(newDegree: Int): Vertex = copy(degree=newDegree)

  override def setTs(newTs: Long): Vertex = copy(ts=newTs)

  def text = ID.lastPart(id).replace("_", " ")

  override def toString: String = text

  override def raw: String = {
    "type: " + "text<br />" +
    "id: " + id + "<br />"
  }
}

object EntityNode {
  def fromNsAndText(namespace: String,
    text: String,
    degree: Int = 0,
    ts: Long = -1) = {

    EntityNode(id(namespace, text), degree, ts)
  }

  def id(namespace: String, text: String) =
    namespace + "/" + ID.sanitize(text).toLowerCase
}

File: core/src/main/java/com/graphbrain/db/EntityNode.java
package com.graphbrain.db;

import java.util.Map;

public class EntityNode extends Vertex {

    @Override
    public VertexType type() {return VertexType.Entity;}

    public EntityNode(String id, int degree, long ts) {
        super(id, degree, ts);
    }

    public EntityNode(String id) {
        this(id, 0, -1);
    }

    public EntityNode(String id, Map<String, String> map) {
        super(id, map);
    }

    public EntityNode() {}

    @Override
    public Vertex copy() {
        return new EntityNode(id, degree, ts);
    }

    public String text() {
        return ID.lastPart(id).replace("_", " ");
    }

    public static String id(String namespace, String text) {
        return namespace + "/" + ID.sanitize(text).toLowerCase();
    }

    public static EntityNode fromNsAndText(String namespace, String text) {
        return new EntityNode(id(namespace, text));
    }

    @Override
    public String raw() {
        return "type: "
                + "text<br />"
                + "id: "
                + id
                + "<br />";
    }
}